src/data/nat/factorization.lean

```diff
@@ -235,41 +221,49 @@ noncomputable def rec_on_prime_pow {P : ℕ → Sort*} (h0 : P 0) (h1 : P 1)
   | (k+2) := λ hk, begin
     let p := (k + 2).min_fac,
     have hp : prime p := min_fac_prime (succ_succ_ne_one k),
-    let t := (k+2).factorization p,
-    have hpt : p ^ t ∣ k + 2 := pow_factorization_dvd _ _,
-    have ht  : 0 < t := hp.factorization_pos_of_dvd (nat.succ_ne_zero (k + 1)) (min_fac_dvd _),
+    -- the awkward `let` stuff here is because `factorization` is noncomputable (finsupp);
+    -- we get around this by using the computable `factors.count`, and rewriting when we want
+    -- to use the `factorization` API
+    let t := (k+2).factors.count p,
```

Comment:
@eric-wieser: I feel like I remember you saying that the non-comp part of `finsupp.of_multiset` was the fact it was an `add_equiv`, not an equiv. Is it worth making a computable version? (I'm happy with it being primed or whatever)